tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/NN.java

```diff
@@ -0,0 +1,379 @@
+package org.tensorflow.op.core;
+
+import org.tensorflow.DataType;
+import org.tensorflow.Operand;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.op.Op;
+import org.tensorflow.op.Scope;
+import org.tensorflow.op.annotation.Endpoint;
+import org.tensorflow.op.annotation.Operator;
+import org.tensorflow.op.math.*;
+import org.tensorflow.op.nn.raw.SoftmaxCrossEntropyWithLogits;
+import org.tensorflow.op.nn.raw.SparseSoftmaxCrossEntropyWithLogits;
+import org.tensorflow.types.*;
+import org.tensorflow.types.family.TNumber;
+import org.tensorflow.op.dtypes.Cast;
+import org.tensorflow.types.family.TType;
+import org.tensorflow.op.linalg.Transpose;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+@Operator(group = "nn")
+public abstract class NN {
```

Comment:
Can we rename this class to something else? 

Actually it should be located under `org.tensorflow.op.nn`. The packages in `src/main` and `src/gen` overlaps each other "by design", as it is the case with `org.tensorflow.op.core`. So the idea is to put the wrapper directly into the right directory, `org.tensorflow.op.nn` in this case. 

The class can either be split in 3 classes (one per endpoint, i.e. `SoftmaxCrossEntropyWithLogits`, `SparseSoftmaxCrossEntropyWithLogits` or `SigmoidCrossEntropyWithLogits`) or remain in a single class (`CrossEntropy` maybe? or just `Helpers` to reflect what [the one already present](https://github.com/tensorflow/java/blob/master/tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/Helpers.java) in the core package?)

Reply:
I vote for 3 individual classes under `org.tensorflow.op.nn`.

Reply:
The 3 files have been checked-in